src/components/Map/PlantingSiteMap.tsx

```diff
@@ -125,12 +131,13 @@ export default function PlantingSiteMap(props: PlantingSiteMapProps): JSX.Elemen
   useEffect(() => {
     const fetchPlantingSite = () => {
       const sources = new Array<MapSource>();
+      const isTopmostLayer = () => sources.length === 0;
```

Comment:
This is cool :) .. maybe call it 'leaf level'? (it is top most based on the order of populating sources, but leaf level in the map.. which is what we want selected).
Was thinking about the alternative/deterministic check.
Ex. subzone is interactive if monitoring plots are not selected but subzones are selected - in the layers. True, sources would be empty when populating subzones.. because we skipped populating monitoring plots. If we reordered how sources are populated, this would break.. vs a check of which layers are selected.
I like the simplicity of the current code - no harm as it is.
Just thinking out aloud.

Reply:
You're right, having a deterministic check that doesn't depend on ordering would be nicer. I will make this change. Thanks

Reply:
Ugh.. sorry, I did like this approach, thought it was clever. I think it should be ok as long as we add a comment in there to maintain the ordering?

Reply:
Yeah, the deterministic check is kinda gross looking now that i'm trying to write it. If I changed the name of the function to `isFirstLayerAdded` then it would capture the ordering semantics without needing further explanation, would you agree? I can add a comment explaining the significance of the first layer added (it renders on top of the other layers, and that's where we want to pin the interactivity)

Reply:
That sounds good 👍